components/pluralrules/src/rules/ast.rs

```diff
@@ -0,0 +1,436 @@
+//! `AST` provides a set of Syntax Tree Nodes used to store
+//! the output of [`parse`] method that is used in [`test_condition`] method
+//! to evaluate whether a given [`PluralCategory`] should be used.
+//!
+//! # Examples
+//!
+//! ```
+//! use icu_pluralrules::rules::parse;
+//! use icu_pluralrules::rules::ast::*;
+//!
+//! let input = "i = 1";
+//!
+//! let ast = parse(input.as_bytes())
+//!     .expect("Parsing failed.");
+//!
+//! assert_eq!(ast, Condition(Box::new([
+//!     AndCondition(Box::new([
+//!         Relation {
+//!             expression: Expression {
+//!                 operand: Operand::I,
+//!                 modulus: None,
+//!             },
+//!             operator: Operator::Eq,
+//!             range_list: RangeList(Box::new([
+//!                 RangeListItem::Value(
+//!                     Value(1)
+//!                 )
+//!             ]))
+//!         }
+//!     ]))
+//! ])));
+//! ```
+//!
+//! [`PluralCategory`]: ../enum.PluralCategory.html
+//! [`parse`]: ../fn.parse.html
+//! [`test_condition`]: ../fn.test_condition.html
+use std::ops::RangeInclusive;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Rule {
+    pub condition: Condition,
+    pub samples: Option<Samples>,
+}
+
+/// A complete (and the only complete) AST representation of a plural rule. Comprises a vector of AndConditions.
+///
+/// # Examples
+///
+/// All AST nodes can be built explicitly, as seen in the example. However, due to its complexity, it is preferred to build the AST using the parse_plural_rule function.
+///
+/// ```text
+/// "i = 5"
+/// ```
+///
+/// Can be represented by the AST:
+///
+/// ```
+/// use icu_pluralrules::rules::ast::*;
+///
+/// Condition(Box::new([AndCondition(Box::new([Relation {
+///        expression: Expression {
+///            operand: Operand::I,
+///            modulus: None,
+///        },
+///        operator: Operator::Eq,
+///        range_list: RangeList(Box::new([RangeListItem::Value(Value(5))])),
+///    }]))]));
+/// ```
+///
+/// Because they care complete representations, hand-written Conditions can be verified with the assert macro. No other AST nodes can be verified.
```

Comment:
nit: I can't parse "Because they care complete representation", probably there's a typo. :)

Reply:
This was a copy&paste from the old code. I cleaned up the doc a bit!